bin/phase-functions.sh

```diff
@@ -722,6 +722,14 @@ __dyn_install() {
 	if has nostrip ${FEATURES} ${PORTAGE_RESTRICT} || has strip ${PORTAGE_RESTRICT}; then
 		>> DEBUGBUILD
 	fi
+
+	# Prematurely delete WORKDIR in case merge-wait is enabled, to
+	# decrease the space used by portage build directories until the
+	# packages are merged and cleaned.
+	if has merge-wait ${FEATURES} && ! has keepwork ${FEATURES}; then
+		rm -rf "${WORKDIR}"
+	fi
+
 	trap - SIGINT SIGQUIT
```

Comment:
We should do it after this `trap - SIGINT SIGQUIT` since the __abort_install critical section is over.

Reply:
Fair point, fixed.